### PR TITLE
Add ARM, GPU and custom network support 

### PR DESCRIPTION
**Pull request by markspec**
Added several options to enable me to use for my use case:

1. ARM support (use ARM binaries to enable running on t2a VMs) 
2. GPU support (tested with T4, A100 and L4 GPUs)
3. Run with custom networking configuration

### CODE REVIEW

**Comment by ravwojdyla** [comment]
👋 @markspec looks like a nice PR, I'm curious if you have intentionally closed it?

**Comment by markspec** [comment]
@ravwojdyla  I am going to resolve the merge conflicts and then reopen.  Hopefully later this week.

**Comment by ravwojdyla** [comment]
@markspec sounds great, looking forward!

**Comment by markspec** [comment]
--accelerator now aligns with google api. This is not compatible with the former --accelerator_type arg.

File: action.sh [hunk 1]
```diff
@@ -243,8 +299,8 @@ systemctl enable shutdown.service
   if [[ $GH_READY == 1 ]]; then
     echo "✅ ${VM_ID} ready ..."
   else
-    echo "Waited 2 minutes for ${VM_ID}, without luck, deleting ${VM_ID} ..."
-    gcloud --quiet compute instances delete ${VM_ID} --zone=${machine_zone}
+    echo "Waited 5 minutes for ${VM_ID}, without luck, deleting ${VM_ID} ..."
+    # gcloud --quiet compute instances delete ${VM_ID} --zone=${machine_zone}
```
Comment (ravwojdyla) [inline action.sh:303]: Why commented out?

File: action.sh [hunk 2]
```diff
@@ -194,6 +216,8 @@ systemctl enable shutdown.service
     ./svc.sh install && \\
     ./svc.sh start && \\
     gcloud compute instances add-labels ${VM_ID} --zone=${machine_zone} --labels=gh_ready=1
+    # Install at if not installed
+    command -v at >/dev/null 2>&1 ||  apt-get install -y -q at
```
Comment (ravwojdyla) [inline action.sh:220]: #31 has removed the need for `at`

**Review by ravwojdyla — CHANGES_REQUESTED** [review]
Thanks @markspec Looks awesome! Some requests below.

**Comment by ravwojdyla** [comment]
@andodet FYI in case you want to take a look at the GPU handling ^

File: action.sh (hunk 1 above)
Comment (markspec) [reply action.sh]: To aid debugging.  Fixed with [945578f](https://github.com/related-sciences/gce-github-runner/pull/33/commits/945578fb8561353f87bb9984fdee7cc73f803a78).

File: action.sh (hunk 2 above)
Comment (markspec) [reply action.sh]: Removed in [945578f](https://github.com/related-sciences/gce-github-runner/pull/33/commits/945578fb8561353f87bb9984fdee7cc73f803a78)

File: action.sh [hunk 3]
```diff
@@ -205,15 +227,45 @@ systemctl enable shutdown.service
...
+    ${preemptible_flag} \
+    ${no_external_address_flag} \
+    ${subnet_flag} \
+    ${accelerator} \
+    ${maintenance_policy_flag} \
+    --labels=gh_ready=0 \
+    --metadata=startup-script=\"$startup_script\" \
+    && echo "label=${VM_ID}" >> $GITHUB_OUTPUT
+  "
+
```
Comment (andodet) [inline action.sh:268]: Is this a leftover from debugging?

**Comment by andodet** [comment]
GPU config worked fine with the following configs:
```yaml
        uses: ./.github/actions/gce-github-runner/
        with:
          token: ${{ secrets.GH_SA_TOKEN }}
          project_id: ${{ secrets.GCP_PROJECT_ID }}
          service_account_key: ${{ secrets.GCP_SA_KEY }}
          image_project: ubuntu-os-cloud
          image_family: ubuntu-2204-lts
          ephemeral: true
          runner_ver: "2.303.0"
          accelerator: count=1,type=nvidia-tesla-k80
          machine_zone: us-central1-a
```
Not a huge fan of having 2 different settings on the same line in yaml though, maybe splitting `accelerator` and `accelerator_count` on 2 different attributes might make it a bit friendlier to use.

**Comment by ravwojdyla** [comment]
@andodet yea, I don't have a strong opinion, it fallows google "spec" 🤷‍♂️ @markspec wdyt?

**Comment by markspec** [comment]
@ravwojdyla  Following google "spec" gives a higher chance we will will maintain compatibility so is a marginal preference.  @andodet, agreed in that I dislike he google implementation.

**Review by ravwojdyla — APPROVED** [review]

**Pull request merged**